### ISSUE TITLE
Nested generic type Issue

### ISSUE THREAD

**Issue by brianyu0717** [SalomonBrys/Kotson#20]
Hi, just noticed an issue while using nested types e.g. 
```
data class User(val id: String)

val a = gson.fromJson<List<List<User>>>("[\n" +
            "  [\n" +
            "    {\n" +
            "      \"id\": \"u-abcd\"\n" +
            "    }\n" +
            "  ]\n" +
            "]\n")
```

Expecting ```a``` to be of type ```ArrayList<ArrayList<User>>>``` however I'm getting ```ArrayList<ArrayList<LinkedTreeMap<etc... as per Gson>>>``` It looks like it's not getting past the 'first' level generic type.

**Comment by SalomonBrys** [comment]
This has to do with the fact that, in kotlin, `List<T>` is actually `List<out T>`, meaning it has declaration site variance.

- In Java `new TypeToken<List<User>>(){}.getType()` returns `java.util.List<com.you.User>`
- In Kotlin, `object : TypeToken<List<User>>(){}.type` returns `java.util.List<? extends com.you.User>`

This will be corrected in Kotson 2.5.0, thanks for the report ;)

**Comment by SalomonBrys** [comment]
Fixed in `2.5.0`
